### PR TITLE
Add support for PHP Namespaces

### PR DESCRIPTION
**Pull request by lkwdwrd**
Adds support for PHP Namespaces to the parser. When plugins and themes use namespaces there can be collisions in the parsed function names, and anything else that exists within that namespace. This adds general support for Namespaces in the parser. Namespaces are created as a taxonomy so that you can view what is available in each namespace as an archive page.

### CODE REVIEW

File: lib/class-importer.php [hunk 1]
```diff
@@ -596,25 +605,55 @@ public function import_item( array $data, $parent_post_id = 0, $import_ignored =
...
 
 				default:
-					$this->errors[] = "\t" . sprintf( 'Problem inserting/updating post for function "%1$s"', $data['name'], $post_id->get_error_message() );
+					$this->errors[] = "\t" . sprintf( 'Problem inserting/updating post for function "%1$s"', $ns_name, $post_id->get_error_message() );
 			}
 
 			return false;
 		}
 
+		$namespaces = ( ! empty( $data['namespace'] ) ) ? explode( '\\', $data['namespace'] ) : false;
```
Comment (atimmer) [inline lib/class-importer.php]: We should move this to a new function, this function is big enough as is.

File: lib/class-relationships.php [hunk 2]
```diff
@@ -341,15 +342,34 @@ public function wp_parser_ending_import() {
 
 	}
 
+	public function names_to_slugs( $name, $namespace = null ) {
```
Comment (atimmer) [inline lib/class-relationships.php]: Please add documentation.

File: lib/class-relationships.php [hunk 3]
```diff
@@ -341,15 +342,34 @@ public function wp_parser_ending_import() {
...
+		}
+		$names[] = $this->name_to_slug( $name );
+
+		return $names;
+	}
+
 	/**
 	 * Convert a method, function, or hook name to a post slug.
 	 *
+	 * Based on context, we may need to check on two names for a fuction because
```
Comment (atimmer) [inline lib/class-relationships.php]: Move this to `names_to_slugs`

File: lib/class-importer.php (hunk 1 above)
Comment (lkwdwrd) [inline lib/class-importer.php]: We were discussing doing this as a separate PR to start breaking this whole function up in to smaller pieces. Looking at the larger context, I was remembering a different set of functions when discussing it, and agree with the comment. We can do this here, but we should still probably circle back and break out more of this import function into smaller pieces.

Comment (lkwdwrd) [inline lib/class-importer.php]: This has now been broken out into a protected method

**Pull request merged**